beacon-chain/core/execution/upgrade.go

```diff
@@ -88,12 +88,83 @@ func UpgradeToBellatrix(state state.BeaconState) (state.BeaconState, error) {
+		return nil, err
+	}
+	currentEpochParticipation, err := state.CurrentEpochParticipation()
+	if err != nil {
+		return nil, err
+	}
+	inactivityScores, err := state.InactivityScores()
+	if err != nil {
+		return nil, err
+	}
+
+	latestExecutionPayloadHeader, err := state.LatestExecutionPayloadHeader()
+	if err != nil {
 		return nil, err
 	}
-	return state, nil
+
+	// Copy all the old fields and add ExcessBlobs of 0
+	upgradedLatestExecutionPayloadHeader := &enginev1.ExecutionPayloadHeader4844{
+		ParentHash:       latestExecutionPayloadHeader.GetParentHash(),
+		FeeRecipient:     latestExecutionPayloadHeader.GetFeeRecipient(),
+		StateRoot:        latestExecutionPayloadHeader.GetStateRoot(),
+		ReceiptsRoot:     latestExecutionPayloadHeader.GetReceiptsRoot(),
+		LogsBloom:        latestExecutionPayloadHeader.GetLogsBloom(),
+		PrevRandao:       latestExecutionPayloadHeader.GetPrevRandao(),
+		BlockNumber:      latestExecutionPayloadHeader.GetBlockNumber(),
+		GasLimit:         latestExecutionPayloadHeader.GetGasLimit(),
+		GasUsed:          latestExecutionPayloadHeader.GetGasUsed(),
+		Timestamp:        latestExecutionPayloadHeader.GetTimestamp(),
+		BaseFeePerGas:    latestExecutionPayloadHeader.GetBaseFeePerGas(),
+		BlockHash:        latestExecutionPayloadHeader.GetBlockHash(),
+		TransactionsRoot: latestExecutionPayloadHeader.GetTransactionsRoot(),
+		ExcessBlobs:      0,
+	}
+
+	s := &ethpb.BeaconState4844{
+		GenesisTime:           state.GenesisTime(),
+		GenesisValidatorsRoot: state.GenesisValidatorsRoot(),
+		Slot:                  state.Slot(),
+		Fork: &ethpb.Fork{
+			PreviousVersion: state.Fork().CurrentVersion,
+			CurrentVersion:  params.BeaconConfig().Eip4844ForkVersion,
+			Epoch:           epoch,
+		},
+		LatestBlockHeader:           state.LatestBlockHeader(),
+		BlockRoots:                  state.BlockRoots(),
+		StateRoots:                  state.StateRoots(),
+		HistoricalRoots:             state.HistoricalRoots(),
+		Eth1Data:                    state.Eth1Data(),
+		Eth1DataVotes:               state.Eth1DataVotes(),
+		Eth1DepositIndex:            state.Eth1DepositIndex(),
+		Validators:                  state.Validators(),
+		Balances:                    state.Balances(),
+		RandaoMixes:                 state.RandaoMixes(),
+		Slashings:                   state.Slashings(),
+		PreviousEpochParticipation:  prevEpochParticipation,
+		CurrentEpochParticipation:   currentEpochParticipation,
+		JustificationBits:           state.JustificationBits(),
+		PreviousJustifiedCheckpoint: state.PreviousJustifiedCheckpoint(),
+		CurrentJustifiedCheckpoint:  state.CurrentJustifiedCheckpoint(),
+		FinalizedCheckpoint:         state.FinalizedCheckpoint(),
+		InactivityScores:            inactivityScores,
+		CurrentSyncCommittee:        currentSyncCommittee,
+		NextSyncCommittee:           nextSyncCommittee,
+		LatestExecutionPayloadHeader: upgradedLatestExecutionPayloadHeader,
+	}
+
+	return state_native.InitializeFromProtoUnsafe4844(s)
```

Comment:
I am creating an entirely new BeaconState -- simply because the LatestExecutionPayloadHeader type has changed (to include ExcessBlobs).